packages/at_client/test/sync_new_test.dart

```diff
@@ -0,0 +1,1922 @@
+      expect(commitEntryResult!.operation, CommitOp.DELETE);
+      await TestResources.tearDownLocalStorage();
+    });
+
+    /// Preconditions:
+    /// 1. There should be an entry for the same key in the key store
+    /// 2. There should be an entry for the same key in the commit log
+
+    // Operation
+    /// Delete a key and insert the same key again
+
+    // Assertions :
+    /// 1. Keystore should have the self key with the new value inserted
+    /// 2. CommitLog should have a following entries in sequence as described below
+    ///     a. Commit entry with CommitOp.Delete
+    ///     b. CommitEntry with CommitOp.Update
+    test('Verify uncommitted queue on re-creation of a self key', () async {
+      //------------Setup---------------------------------
+      await TestResources.setupLocalStorage(atsign);
+      HiveKeystore? keystore = TestResources.getHiveKeyStore(atsign);
+      var atData = AtData();
+      atData.data = 'alice@gmail.com';
+      var newData = AtData();
+      newData.data = 'alice@yahoo.com';
+      //------------Preconditions SetUp---------------------------------
+      String atKey = (AtKey.self('email', namespace: 'wavi', sharedBy: atsign))
+          .build()
+          .toString();
+      await keystore!.put(atKey, atData);
+      // verifying the key in the commit log
+      final commitLogInstance =
+          await (AtCommitLogManagerImpl.getInstance().getCommitLog(atsign));
+      // seq number from the commit entry
+      var seqNum = commitLogInstance!.lastCommittedSequenceNumber();
+      //-----------Operation---------------------------------
+      // remove the created key
+      await keystore.remove(atKey);
+      // re-creating the same key in the keystore with a different value
+      await keystore.put(atKey, newData);
+      //-----------Assertions---------------------------------:
+      // verifying the key in the key store
+      var keyStoreGetResult = await keystore.get(atKey);
+      expect(keyStoreGetResult!.data, 'alice@yahoo.com');
+      // verifiyng the createdAt time is less than DateTime.now()
+      expect(keyStoreGetResult.metaData!.updatedAt!.isBefore(DateTime.now()),
+          true);
+      // verify the entries in the commit log
+      var commitEntriesResult = await SyncUtil()
+          .getChangesSinceLastCommit(seqNum, 'wavi', atSign: atsign);
+      expect(commitEntriesResult[0].operation, CommitOp.DELETE);
+      expect(commitEntriesResult[1].operation, CommitOp.UPDATE);
+      await TestResources.tearDownLocalStorage();
+    });
+
+    /// Preconditions
+    /// 1. There should be no entry for the same key in the key store
+    /// 2. There should be no entry for the same key in the commit log
+
+    // Operation
+    /// Put a local key
+
+    // Assertions
+    /// 1. Keystore should have the local key with the value inserted
+    /// 2. Assert the metadata of the key. "CreatedAt" should be populated with
+    /// DateTime which is less than DateTime.now()
```

Comment:
Assert that the local key should not be present in the commit log. Pls, add it to the docs and code.